general-doc/decentralized-web.rst

```diff
@@ -19,25 +19,20 @@ Popular ETH smart contracts essentially follow that rule.
 
 However, most DeFi web3 apps such as `app.uniswap.org <https://app.uniswap.org/>`_ are
 typically used through an otherwise normal website that talks to a browser plugin
-(typically `Metamask <https://github.com/MetaMask/metamask-extension>`_) allowing the webpage
-to interact with the user's wallet and the blockchain. The website that serves as an
-entry point to the dApp is neither decentralized nor immutable-once-audited, thus breaking
-the very foundations of DeFi security. And that’s how you get into situations of
-scandalous thefts in the DeFi world, like
-`this one <https://www.theverge.com/2021/12/2/22814849/badgerdao-defi-120-million-hack-bitcoin-ethereum>`_.
+(typically `Metamask <https://github.com/MetaMask/metamask-extension>`_) allowing the webpage to interact with the user's wallet and the blockchain. The website that serves as an
+entry point to the dApp is neither decentralized nor immutable-once-audited.
+This breaks the very foundations of blockchain security. And that’s how you get into situations of scandalous thefts in the DeFi world, like `this one <https://www.theverge.com/2021/12/2/22814849/badgerdao-defi-120-million-hack-bitcoin-ethereum>`_.
 
 The goal here is to allow addresses on Massa to store not only a balance, bytecode and a
-datastore, but also files. Those files must remain small, or storing them will cost a lot to their owner.
+datastore, but also files without using any centralised party in between your client and the blockchain.
 Any address, through bytecode execution, can initialize, read and write the "files" part
 just like it would with the datastore. The reason why we don’t reuse the datastore for this,
 outside of the risk of key collisions, is for easier auditing: if the code never writes
 into its own bytecode nor its filestore after deployment, it is safe to assume that the stored
-website can’t change anymore.
+website can’t change anymore. For that Massa has developed a client that acts as a gateway to the blockchain preventing you from using any centralised servers but also maximising your security with immutable and censorship resistant websites.
 
-In particular, Massa can host the files corresponding to the front-end
```

Comment:
I wouldn't remove this section since it gives a good explanation that it's safe

Reply:
which section? Sorry I don't see what I removed here.

Reply:
"Massa can host the files corresponding to the front-end of the decentralized applications. Since the front-end is hosted on the blockchain, it becomes de facto immutable and censorship resistant, allowing anyone to access it using a Massa node. That way, Massa allows deploying fully decentralized code-is-law apps, as it was meant to be!" 

I'd return that section before "For that Massa developed...."